3-UsingAPIs/Week2/assignment/ex2-pokemonApp/index.js

```diff
@@ -22,18 +23,86 @@ Use async/await and try/catch to handle promises.
 Try and avoid using global variables. As much as possible, try and use function 
 parameters and return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchData(url) {
+  try {
+    const response = await fetch(url);
+    if (!response.ok) {
+      throw new Error('HTTP or network errors');
+    }
+    const pokemon = await response.json();
+    return pokemon;
+  } catch (error) {
+    console.log('HTTP or network errors');
+  }
 }
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+// eslint-disable-next-line no-unused-vars
+async function fetchAndPopulatePokemons() {
+  const contentDiv = document.querySelector('.content');
+  const select = document.querySelector('.select-pokemon');
+  console.log('select', select);
+  try {
+    const pokemons = await fetchData(
+      'https://pokeapi.co/api/v2/pokemon?limit=151'
+    );
+
+    pokemons.results.forEach((pokemon) => {
+      const option = document.createElement('option');
+      option.textContent = pokemon.name;
+      option.value = pokemon.url;
+      select.appendChild(option);
+    });
+
+    contentDiv.appendChild(select);
+  } catch (error) {
+    console.error('Error fetching and populating pokemons:', error);
+  }
 }
 
-function fetchImage(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+// eslint-disable-next-line no-unused-vars
+async function fetchImage() {
+  const selectedPokemonUrl = document.querySelector('.select-pokemon').value;
```

Comment:
It is customary to use an event argument to pass information from user interactions. See my comment later in your `main()` function.

```suggestion
async function fetchImage(event) {
  const selectedPokemonUrl = event.target.value;